### PR TITLE
CTP-2346 code review

### PR DESCRIPTION
**Pull request by aydevworks**
Related to https://ucldata.atlassian.net/browse/CTP-2346

Just some suggestions. May be add a README.md file to describe what this plugin does.

### CODE REVIEW

File: block_pickup.php [hunk 1]
```diff
@@ -39,8 +41,8 @@ public function init() {
      *
      */
     public function specialization() {
-        if (isset($this->config->title)) {
-            $this->title = $this->title = format_string($this->config->title, true, ['context' => $this->context]);
+        if (!empty($this->config->title)) {
```
Comment (aydevworks) [inline block_pickup.php:44]: More accurate to use empty() to check for empty string

File: block_pickup.php [hunk 2]
```diff
@@ -85,30 +87,13 @@ public function get_content() : stdClass {
...
-        $params = array(
-            'userid' => $USER->id,
-        );
-        $modrecords = $DB->get_records_sql($sql, $params);
-
-        if (!count($modrecords)) {
-            return array();
-        }
+        /* Get the recent items using recentlyaccesseditems block's helper class */
+        $modrecords = helper::get_recent_items(4);
```
Comment (aydevworks) [inline block_pickup.php:91]: Recommend to use the existing helper class to get the recently accessed items.

File: block_pickup.php [hunk 3]
```diff
@@ -85,30 +87,13 @@ public function get_content() : stdClass {
...
 
         /* Template data for mustache. */
         $template = new stdClass();
 
         foreach ($modrecords as $cm) {
-            $contextmodule = context_module::instance($cm->cmid);
```
Comment (aydevworks) [inline block_pickup.php:111]: `$contextmodule ` is unused.

File: block_pickup.php [hunk 4]
```diff
@@ -132,21 +117,10 @@ public function fetch_recent_mods() : array {
...
-                 LIMIT 3";
-
-        $params = array(
-            'userid' => $USER->id,
-        );
-        $courserecords = $DB->get_records_sql($sql, $params);
+        global $USER;
+
+        // Get recent courses.
+        $courserecords = course_get_recent_courses($USER->id, 3);
```
Comment (aydevworks) [inline block_pickup.php:127]: Suggest to use moodle core function to get the recent courses.

File: block_pickup.php [hunk 5]
```diff
@@ -160,29 +134,16 @@ public function fetch_recent_courses() : array {
...
 
             /* Progress. */
-            if ($cr->enablecompletion) {
-                $percentage = \core_completion\progress::get_course_progress_percentage($cr, $USER->id);
-                if (!is_null($percentage)) {
-                    $percentage = floor($percentage);
-                    $course->progress = $percentage;
-
-                }
+            if ($percentage = progress::get_course_progress_percentage($cr, $USER->id)) {
```
Comment (aydevworks) [inline block_pickup.php:144]: `enablecompletion ` is checked inside the `\core_completion\progress::get_course_progress_percentage` function, so no need to check it before.

File: block_pickup.php [hunk 6]
```diff
@@ -160,29 +134,16 @@ public function fetch_recent_courses() : array {
...
-
             /* Course image. */
-            foreach ($cle->get_course_overviewfiles() as $file) {
-                $course->courseimage = file_encode_url("$CFG->wwwroot/pluginfile.php",
-                '/' . $file->get_contextid() .
-                '/' . $file->get_component() .
-                '/' . $file->get_filearea() . $file->get_filepath() . $file->get_filename());
-            }
-
+            $course->courseimage = course_summary_exporter::get_course_image($cr);
```
Comment (aydevworks) [inline block_pickup.php:150]: use core function to get course image and `file_encode_url` is deprecated.

File: lang/en/block_pickup.php [hunk 7]
```diff
@@ -28,11 +28,4 @@
 $string['privacy:metadata'] = 'Pick up where you left off does not store any personal data.';
 
 $string['blocktitle'] = 'Block title';
-$string['configurealerts'] = "Edit global alert posts";
```
Comment (aydevworks) [inline lang/en/block_pickup.php:31]: removed unused strings

File: version.php [hunk 8]
```diff
@@ -25,6 +25,8 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->version   = 2023060100;        // The current plugin version (Date: YYYYMMDDXX).
+$plugin->release   = '1.0';
```
Comment (aydevworks) [inline version.php:29]: Looks like more completed version.php
https://moodledev.io/docs/apis/commonfiles/version.php

**Review by stuartlamour — APPROVED** [review]

**Review by aspark21 — CHANGES_REQUESTED** [review]
there's something that needs looking at with regards to recently accessed activities

talk next week

File: block_pickup.php (hunk 1 above)
Comment (stuartlamour) [reply block_pickup.php:44]: No title ever on this block.

**Comment by aydevworks** [comment]
> Member

There is no recently accessed items shown when logged in as other user. I will have a look of it.

**Comment by aydevworks** [comment]
> > Member
> 
> There is no recently accessed items shown when logged in as other user. I will have a look of it.

This is to prevent access tracking: https://github.com/moodle/moodle/blob/MOODLE_402_STABLE/blocks/recentlyaccesseditems/classes/helper.php#L51-L54

**Comment by aydevworks** [comment]
Rebased against main branch.

**Review by aspark21 — APPROVED** [review]

**Pull request merged**